src/components/ContentTemplate.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+	<NcAppContent :show-details="showDetails"
+		@update:showDetails="unselectItem()">
+		<template #list>
+			<NcAppContentList>
+				<div class="header">
+					<slot name="header" />
+				</div>
+
+				<FeedItemDisplayList :items="items"
+					:fetch-key="fetchKey"
+					:config="config"
+					@load-more="emit('load-more')" />
+			</NcAppContentList>
+		</template>
+
+		<NcAppContentDetails class="content-details-container">
+			<FeedItemDisplay v-if="selectedFeedItem" :item="selectedFeedItem" />
+		</NcAppContentDetails>
+	</NcAppContent>
+</template>
+
+<script setup lang="ts">
+
+import { PropType, computed, ref, watch } from 'vue'
+
+import itemStore from '../store/item'
+
+import NcAppContent from '@nextcloud/vue/dist/Components/NcAppContent'
+import NcAppContentList from '@nextcloud/vue/dist/Components/NcAppContentList'
+import NcAppContentDetails from '@nextcloud/vue/dist/Components/NcAppContentDetails'
+
+import { FeedItem } from '../types/FeedItem'
+
+import FeedItemDisplayList from './feed-display/FeedItemDisplayList.vue'
+import FeedItemDisplay from './feed-display/FeedItemDisplay.vue'
+
+defineProps({
```

Comment:
I'm not up to date on the latest Vue/Nextcloud component creation patterns, is this the preferred way to do this now? 

the other components I wrote used `Vue.extend`

Reply:
I'm using composition api since in my opinion it creates much cleaner code and is the way to go in the future.

Reply:
Ah, I have heard of it. I probably should have implemented the other components using that /: 

makes sense though, I have heard that it is the direction Vue is moving

Reply:
Yeah, but options api was used for a long time so you probably got used to it like many other people.. I started with vue rather recently so I decided to accomodate myself to it from the start.. 😉

Reply:
I see, that makes sense

Unfortunate that it looks completely different from the other files, and we can't fully adopt it correctly with Vuex v4 because of Nextclouds dependency. But it probably makes better sense to use the `NcAppContentList` and `NcAppContentDetails` components

Reply:
Yes, but that's more a problem of Vuex because it doesn't want to support composition api for vue 2.7. Which I also can understand as a valid point because they would have to maintain another combination

Reply:
Also, as far as I'm informed Nextcloud will switch to Vue 3 since support will end with 2023. So we can deal with it later..